lib/membrane/rtp/twcc_sender/receiver_rate.ex

```diff
@@ -58,9 +60,12 @@ defmodule Membrane.RTP.TWCCSender.ReceiverRate do
     receive_deltas
     |> Enum.zip(packet_sizes)
     |> Enum.filter(fn {delta, _size} -> delta != :not_received end)
-    |> Enum.scan({reference_time, List.first(packet_sizes)}, fn {recv_delta, size},
-                                                                {prev_timestamp, _prev_size} ->
-      {prev_timestamp + recv_delta, size}
+    |> Enum.reduce({reference_time, :queue.new()}, fn {recv_delta, size},
+                                                      {prev_timestamp, packets_received} ->
+      receive_timestamp = prev_timestamp + recv_delta
+      {receive_timestamp, :queue.in({receive_timestamp, size}, packets_received)}
```

Comment:
Since you operate on a list either way, maybe it would more readable to use map_reduce and (after `elem`) using `:queue.from_list`?

Reply:
I have changed that but the main difference was that I was creating final queue in one pass so in O(n) while now it is created in 2 * O(n) as :queue.from_list/Qex.new is O(n)